azurelinuxagent/common/protocol/util.py

```diff
@@ -74,6 +74,7 @@ class ProtocolUtil(SingletonPerThread):
     """
 
     def __init__(self):
+        self._lock = threading.RLock()  # protects the files on disk created during protocol detection
```

Comment:
What is the idea behind using a reentrant thread lock instead of a regular thread lock?